core/src/main/java/com/whylogs/core/ColumnProfile.java

```diff
@@ -39,20 +42,42 @@
   @NonNull private final ItemsSketch<String> frequentItems;
   @NonNull private final HllSketch cardinalityTracker;
 
+  static void initNullCheck() {
+    //  "NULL_STRINGS" env var specifies custom values interpreted as null, e.g. "nil.NaN,nan,null"
+    //  Missing value (actual null object) is always included.
+    if (ColumnProfile.nullStrs == null) {
+      ColumnProfile.initNullCheck(System.getenv("NULL_STRINGS"));
+    }
+  }
+
+  static void initNullCheck(String nullSpec) {
+    //  Customize null values, e.g. "nil.NaN,nan,null"; Missing value (actual null object) is always
+    // included.
+    ColumnProfile.nullStrs = nullSpec == null ? null : Arrays.asList(nullSpec.split(","));
+  }
+
+  static boolean nullCheck(Object v) {
+    return v == null
+        || (ColumnProfile.nullStrs != null
+            && ((v instanceof String && ColumnProfile.nullStrs.contains(v))
+                || ColumnProfile.nullStrs.contains(v.toString())));
+  }
+
   public ColumnProfile(String columnName) {
     this.columnName = columnName;
     this.counters = new CountersTracker();
     this.schemaTracker = new SchemaTracker();
     this.numberTracker = new NumberTracker();
     this.frequentItems = FrequentStringsSketch.create();
     this.cardinalityTracker = new HllSketch(CARDINALITY_LG_K);
+    ColumnProfile.initNullCheck();
```

Comment:
This is not recommended since you're running this code over and over ever time a ColumnProfile is initiated. Plus it makes it hard to test since the static variable is dependent on the environment variable and initialize at construction time.

A better pattern would be to have another package-level constructor:

```
ColumnProfile(String columnName, List<String> nullStrs) 
```

and the default constructor can do something like:

```
private static List<String> ENV;

  static List<String> nullStrsFromEnv() {
... // logic here
  }

public ColumnProfile(String columnName) {
   this(columnName, nullStrsFromEnv())
}
```

Reply:
I think it would be ideal if we could initialize the static null list from the environment only once.
My code admittedly did not do that correctly but calling nullStrsFromEnv() at every instance constructor seems wrong.  Do you agree the null list should be initialized once and only once? Its not like we have a different null list for each column.

Reply:
Also I anticipated the difficulty of testing if the value depends on environment variable,
that's why I added `static void initNullCheck(String nullSpec)`
which is used by the unit test I added.

Reply:
The code path is leaky though. It means that if one test runs the code path, the other test calls in `initNullCheck` in two different locations it affects the tests - which means the tests are not self-contained. It should be avoided at all cost.

> Do you agree the null list should be initialized once and only once? Its not like we have a different null list for each column.

yes, initialized once, but you can pass the immutable list around rather than having a static variable. That way it only costs a pointer :)